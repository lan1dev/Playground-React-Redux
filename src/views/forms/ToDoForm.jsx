import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Button from '../common/Button';
import InputField from '../common/InputField';
import FlexBox from '../common/FlexBox';

const propTypes = {
  addToDo: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

const validate = values => {
  const errors = {};
  if (!values.todo) {
    errors.todo = 'You need to enter a name for your todo!';
  }
  return errors;
};

class ToDoForm extends Component {
  handleSubmit = values => {
    this.props.addToDo(values.todo);
  };

  render() {
    return (
      <form onSubmit={this.props.handleSubmit(this.handleSubmit)}>
        <FlexBox>
          <Field name="todo" label="Add ToDo" component={InputField} type="text" />
          <Button type="submit">Add</Button>
        </FlexBox>
      </form>
    );
  }
}

ToDoForm.propTypes = propTypes;

/* eslint-disable no-class-assign */
ToDoForm = reduxForm({
  form: 'addToDo',
  warn: () => {},
  validate
})(ToDoForm);

export default ToDoForm;
