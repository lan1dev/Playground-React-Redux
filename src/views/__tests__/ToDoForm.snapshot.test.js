import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ToDoForm from '../forms/ToDoForm';

describe('ToDoForm', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<ToDoForm {...props} />);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: undefined
    };
    wrapper = undefined;
  });

  describe('when component has rendered', () => {
    props = {
      addToDo: jest.fn(),
      handleChange: jest.fn(),
      addToDo: ''
    };
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
