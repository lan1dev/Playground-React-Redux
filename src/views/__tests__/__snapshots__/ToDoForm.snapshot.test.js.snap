// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoForm when component has rendered Should match snapshot 1`] = `
<Connect(Form(ToDoForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="addToDo"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
  warn={[Function]}
/>
`;
